src/gui/editorwidgets/qgsqmlwidgetwrapper.h

```diff
@@ -0,0 +1,69 @@
+/***************************************************************************
+  qgsqmlwidgetwrapper.h
+
+ ---------------------
+ begin                : 25.6.2018
+ copyright            : (C) 2018 by Matthias Kuhn
+ email                : matthias@opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSQMLWIDGETWRAPPER_H
+#define QGSQMLWIDGETWRAPPER_H
+
+#include "qgswidgetwrapper.h"
+#include "qgis.h"
+#include "qgis_gui.h"
+#include <QtQuickWidgets/QQuickWidget>
+
+class GUI_EXPORT QgsQmlWidgetWrapper : public QgsWidgetWrapper
+{
+    Q_OBJECT
+
+  public:
+
+    QgsQmlWidgetWrapper( QgsVectorLayer *layer, QWidget *editor, QWidget *parent );
+
+    bool valid() const override;
+
+    QWidget *createWidget( QWidget *parent ) override;
+
+    void initWidget( QWidget *editor ) override;
+
+    //! Clears the content and makes new intialisaton
+    void reinitWidget();
+
+    //! writes the \a qmlCode into a temporary file
+    void setQmlCode( const QString &qmlCode );
+
+  public slots:
+
+    //! passes the \a feature into the context property of the widget
+    void setFeature( const QgsFeature &feature ) override;
+
+  private:
+    QTemporaryFile mQmlFile;
+    QQuickWidget *mWidget = nullptr;
+};
+
+
+class GUI_EXPORT QmlExpression : public QObject
```

Comment:
can this be put into the cpp file?
I would keep this as an implementation detail (and not public api)